### PR TITLE
#33: Extracted and simplified SelectedArticle from SDK to UI

### PR DESCRIPTION
**Pull request by bonkey**

### CODE REVIEW

File: AtlasSDK/AtlasSDK.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1113,7 +1109,7 @@
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				COPY_PHASE_STRIP = NO;
-				CURRENT_PROJECT_VERSION = 1.1.5;
+				CURRENT_PROJECT_VERSION = 1.1.6;
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK.xcodeproj/project.pbxproj]: Will fix it...

**Comment by bonkey** [comment]
:+1:

File: AtlasUI/AtlasUI/Utils/SelectedArticle.swift [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+func * (lhs: Decimal, rhs: Int) -> Decimal {
+    return lhs * Decimal(rhs)
+}
+
+func == (lhs: SelectedArticle, rhs: SelectedArticle) -> Bool {
+    return lhs.article.id == rhs.article.id && lhs.unit == rhs.unit && lhs.quantity == rhs.quantity
+}
+
+func != (lhs: SelectedArticle, rhs: SelectedArticle) -> Bool {
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/Utils/SelectedArticle.swift]: where these are used?

Comment (bonkey) [inline AtlasUI/AtlasUI/Utils/SelectedArticle.swift]: eg. https://github.com/zalando-incubator/atlas-ios/blob/master/AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift#L83

File: AtlasUI/AtlasUI/CheckoutSummaryProductInfoStackView.swift [hunk 3]
```diff
@@ -89,11 +89,7 @@ extension CheckoutSummaryProductInfoStackView: UIDataBuilder {
         brandNameLabel.text = viewModel.article.brand.name
         articleNameLabel.text = viewModel.article.name
         unitColorLabel.text = viewModel.article.color
-        if let price = viewModel.unit?.price {
-            unitPriceLabel.text = Localizer.format(price: price)
-        } else {
-            unitPriceLabel.text = ""
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryProductInfoStackView.swift]: ut6xbl4kfdcwc

File: AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift [hunk 4]
```diff
@@ -66,7 +66,7 @@ class CheckoutSummaryEditProductStackView: UIStackView {
     }
 
     private func sizeSelected(at idx: Int, for selectedArticle: SelectedArticle) {
-        let updatedArticle = SelectedArticle(article: selectedArticle.article, desiredQuantity: selectedArticle.quantity, unitIndex: idx)
+        let updatedArticle = SelectedArticle(in: selectedArticle, changeSelectedIndex: idx)
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryEditProductStackView.swift]: Well, was like that, but it's like `changeSelectedIndex:` `in`, maybe reorder it...

File: AtlasSDK/AtlasSDK/AtlasAPI.swift [hunk 5]
```diff
@@ -48,18 +48,18 @@ extension AtlasAPI {
         client.fetch(from: endpoint, completion: completion)
     }
 
-    public func createCheckoutCart(for selectedArticle: SelectedArticle,
+    public func createCheckoutCart(for skuQuantity: SKUQuantity,
                                    addresses: CheckoutAddresses? = nil,
                                    completion: @escaping APIResultCompletion<CartCheckout>) {
-        let cartItemRequest = CartItemRequest(sku: selectedArticle.sku, quantity: selectedArticle.quantity)
+        let cartItemRequest = CartItemRequest(sku: skuQuantity.sku, quantity: skuQuantity.quantity)
```
Comment (bonkey) [inline AtlasSDK/AtlasSDK/AtlasAPI.swift]: Well, maybe...

File: AtlasUI/AtlasUI/Utils/SelectedArticle.swift (hunk 2 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/Utils/SelectedArticle.swift]: ah it's old one, actually I meant these 
https://github.com/zalando-incubator/atlas-ios/pull/765/files#diff-a6f108caf677be027bf6b08074d31ec6R166

File: AtlasUI/AtlasUI/CheckoutSummaryProductInfoStackView.swift (hunk 3 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/CheckoutSummaryProductInfoStackView.swift]: sn6cz79tz53rs


Look at the `€0,00` 😎

simulator screen shot 28 feb 2017 6 33 47 pm

File: AtlasUI/AtlasUI/Utils/SelectedArticle.swift (hunk 2 above)
Comment (bonkey) [inline AtlasUI/AtlasUI/Utils/SelectedArticle.swift]: Yeah, cleaned it. I think so, as this link points to removed code, too.

**Pull request merged**